policy-engine/src/openapi.rs

```diff
@@ -0,0 +1,65 @@
+use actix_web::{HttpRequest, HttpResponse};
+use openapiv3::OpenAPI;
+use AppState;
+
+const SPEC: &str = include_str!("openapiv3.json");
+
+pub(crate) fn index(req: HttpRequest<AppState>) -> HttpResponse {
+    let path_prefix = &req.state().path_prefix;
+
+    let mut spec_object: OpenAPI = match serde_json::from_str(SPEC) {
+        Ok(o) => o,
+        Err(e) => {
+            let e = format_err!("Could not deserialize to OpenAPI object: {}", e);
+            error!("{}", e);
+            return HttpResponse::from_error(e.into());
+        }
+    };
+
+    // Prefix all paths with `path_prefix`
+    spec_object.paths = rewrite_paths(&spec_object.paths, path_prefix);
+
+    match serde_json::to_string(&spec_object) {
+        Ok(s) => HttpResponse::from(s),
+        Err(e) => {
+            let e = format_err!("Could not serialize OpenAPI object: {}", e);
+            error!("{}", e);
+            HttpResponse::from_error(e.into())
+        }
+    }
+}
+
+fn rewrite_paths(paths: &openapiv3::Paths, path_prefix: &str) -> openapiv3::Paths {
+    paths
+        .iter()
+        .map(|(path, path_item)| {
+            let new_path = format!("{}{}", path_prefix, &path);
+            trace!("Rewrote path {} -> {} ", &path, &new_path);
+            (new_path, path_item.clone())
+        })
+        .collect()
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Can we maybe add a roundtrip test to make sure the library is not returning crazy results at runtime?

Reply:
I think that'd be redundantly testing the crates we're using which I'd avoid if possible. I did test it manually and it works as expected, and our unit test covers any OpenAPI related mistake which could be made in the spec itself.

Reply:
Ack, I guess I was being over-cautious there :)